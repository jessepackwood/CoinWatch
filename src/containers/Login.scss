.login {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.email-input, .password-input {
  height: 50px;
  width: 400px;
  margin: 10px;
}

.btn-submit {
  width: 400px;
  height: 50px;
  margin: 10px;
}