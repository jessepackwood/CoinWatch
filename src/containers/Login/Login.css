@import url("https://fonts.googleapis.com/css?family=Abel");
.login-form {
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: center;
  align-items: center;
  background: #244968; }

.input-field,
.btn-submit {
  margin-top: 20px;
  height: 35px;
  width: 300px;
  font-size: 14pt; }

.btn-submit {
  margin-bottom: 30px;
  color: #fff;
  background-color: #244968;
  border: 1px solid #fff;
  outline: none; }
  .btn-submit:hover {
    cursor: pointer;
    border: 1px solid #9af765; }

.modal-title {
  margin-top: 20px;
  font-size: 16pt; }

.error-message {
  margin-top: 20px;
  color: #fff; }
