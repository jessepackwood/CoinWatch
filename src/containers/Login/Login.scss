@import '../../variable.scss';

// .login {
//   display: flex;
//   flex-direction: column;
//   justify-content: center;
//   align-items: center;
//   background: #434343;
// }

// .email-input, .password-input {
//   height: 25px;
//   width: 400px;
//   margin: 10px;
// }

// .btn-submit {
//   width: 400px;
//   height: 50px;
//   margin: 10px;
// }

.login {
  background: $blue;
  border-radius: 20px;
  // color: white;
  display: flex;
  height: auto;
  width: 400px;
  position: fixed;
  top: calc(50% - 150px);
  left: calc(50% - 200px);
  flex-direction: column;
  justify-content: center;
  align-items: center;
  z-index: 1;
}

.input-field,
.btn-submit {
  margin-top: 20px;
  height: 30px;
  width: 75%;
  font-size: 14pt;

}

.btn-submit {
  margin-bottom: 30px;
  color: #fff;
  background-color: $blue;
  border: none;
}

.modal-title {
  margin-top: 20px;
  font-size: 16pt;
}