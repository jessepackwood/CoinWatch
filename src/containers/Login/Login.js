import React, { Component } from 'react'
import { Redirect } from 'react-router-dom'
import { connect } from 'react-redux'
import * as actions from '../../actions'
// import { googleSignIn } from '../../services/firebase'
import Form from '../../components/Form/Form'
import PropTypes from 'prop-types'
import './Login.css'


class Login extends Component {

  handleInputChange = (event) => {
    const {name, value} = event.target
    this.props.inputChange(name, value)
  }

  handleSubmit = () => {
    this.props.loginUser(this.props.email, this.props.password)
    // googleSignIn().then( (data)=> {
    //   console.log(data)
    // })
  }

  render() {
    return (
      <div>
        { this.props.loggedIn && 
          <Redirect to='/home' />
        }
        <Form 
          email={this.props.email}
          password={this.props.password}
          inputChange={this.handleInputChange}
          submit={this.handleSubmit}
          error={this.props.error}
        />
      </div>
    )
  }
}

export const mapStateToProps = state => {
  return {
    email: state.user.email,
    password: state.user.password,
    error: state.user.error,
    loggedIn: state.user.loggedIn
  }
}

export const mapDispatchToProps = dispatch => {
  return {
    inputChange: (name, value) => {
      dispatch(actions.inputChange(name, value));
    },
    loginUser: (email, password) => {
      dispatch(actions.loginUser(email, password));
    },
    createUser: (email, password) => {
      dispatch(actions.createUser(email, password))
    }
  };
};

Login.propTypes = {
  email: PropTypes.string,
  password: PropTypes.string,
  error: PropTypes.string,
  loggedIn: PropTypes.string,
  inputChange: PropTypes.func,
  loginUser: PropTypes.func,
  createUser: PropTypes.func
}

export default connect(mapStateToProps, mapDispatchToProps)(Login)