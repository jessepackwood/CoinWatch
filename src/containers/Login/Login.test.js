import React from 'react'
import Login from './Login'
import { shallow } from 'enzyme'
import mapStateToProps from './Login'

describe('Login tests', () => {
	it('should exist', () => {
		const login = shallow(<Login />)
		expect(login).toBeDefined()
	})

	it('should receive state from props', () => {
		
	})

	it('should call handleInputChange', () => {

	})


})