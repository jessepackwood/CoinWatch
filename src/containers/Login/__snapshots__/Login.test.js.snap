// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Container Login should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    createUser={[Function]}
    inputChange={[Function]}
    loginUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <div
          className="login-form"
>
          <Link
                    replace={false}
                    to="/home"
          >
                    <h1
                              className="app-title login-title"
                    >
                              Coin Watch
                    </h1>
          </Link>
          <input
                    className="input-field"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    type="text"
                    value=""
          />
          <input
                    className="input-field"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
          />
          <input
                    className="btn-submit"
                    onClick={[Function]}
                    type="submit"
                    value="Login"
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/home"
>
              <h1
                            className="app-title login-title"
              >
                            Coin Watch
              </h1>
</Link>,
            <input
              className="input-field"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
/>,
            <input
              className="input-field"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
/>,
            false,
            <input
              className="btn-submit"
              onClick={[Function]}
              type="submit"
              value="Login"
/>,
            undefined,
          ],
          "className": "login-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <h1
                className="app-title login-title"
>
                Coin Watch
</h1>,
              "replace": false,
              "to": "/home",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Coin Watch",
                "className": "app-title login-title",
              },
              "ref": null,
              "rendered": "Coin Watch",
              "type": "h1",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "btn-submit",
              "onClick": [Function],
              "type": "submit",
              "value": "Login",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div
            className="login-form"
>
            <Link
                        replace={false}
                        to="/home"
            >
                        <h1
                                    className="app-title login-title"
                        >
                                    Coin Watch
                        </h1>
            </Link>
            <input
                        className="input-field"
                        name="email"
                        onChange={[Function]}
                        placeholder="Email"
                        type="text"
                        value=""
            />
            <input
                        className="input-field"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
            <input
                        className="btn-submit"
                        onClick={[Function]}
                        type="submit"
                        value="Login"
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/home"
>
                <h1
                                className="app-title login-title"
                >
                                Coin Watch
                </h1>
</Link>,
              <input
                className="input-field"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
                value=""
/>,
              <input
                className="input-field"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
/>,
              false,
              <input
                className="btn-submit"
                onClick={[Function]}
                type="submit"
                value="Login"
/>,
              undefined,
            ],
            "className": "login-form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <h1
                  className="app-title login-title"
>
                  Coin Watch
</h1>,
                "replace": false,
                "to": "/home",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Coin Watch",
                  "className": "app-title login-title",
                },
                "ref": null,
                "rendered": "Coin Watch",
                "type": "h1",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "btn-submit",
                "onClick": [Function],
                "type": "submit",
                "value": "Login",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
